### ISSUE TITLE
Tags: How does tag context propagation work?

### ISSUE THREAD

**Issue by songy23** [census-instrumentation/opencensus-python#255]
Kindly correct me if I'm wrong:

We added the binary encoding for Tags in https://github.com/census-instrumentation/opencensus-python/pull/209. However, I haven't seen `tag_map` propagation implemented?

**Comment by zeako** [comment]
can you elaborate @songy23 ?
can add support for it

**Comment by songy23** [comment]
This means that we should be able to put a `tag_map` into current `context`, and read the `tag_map` from current `context`. This allows `tag_map` to be propagated on the wire. See how [Go](https://github.com/census-instrumentation/opencensus-go/blob/master/tag/context.go) and [Java](https://github.com/census-instrumentation/opencensus-java/blob/3c74cea59b7c50c65e50401305e7f7feab614a90/impl_core/src/main/java/io/opencensus/implcore/tags/CurrentTagContextUtils.java) did this.

For the definition of `context`, please see https://github.com/census-instrumentation/opencensus-specs#context.

I need to understand how context propagation works in Python a bit - @liyanhui1228 how do we handle setting and reading `trace context`? Probably the same would apply to `tag_map`.

**Comment by liyanhui1228** [comment]
@songy23 trace context is in [here](https://github.com/census-instrumentation/opencensus-python/blob/master/opencensus/trace/execution_context.py). We use the `threading.local()` to store the `span` and `attr` in the context.

**Comment by songy23** [comment]
Thanks, sent https://github.com/census-instrumentation/opencensus-python/pull/257 to fix.

**Comment by colincadams** [comment]
In order to use this, does a user need to explicitly call `set_current_tag_map()`?

**Comment by songy23** [comment]
Yes, currently `set_current_tag_map` is the only way to put a tag_map to current context.
